jetbrains/src/main/java/com/boundaryml/jetbrains_ext/BamlLanguageServerInstaller.java

```diff
@@ -0,0 +1,98 @@
+package com.boundaryml.jetbrains_ext;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.redhat.devtools.lsp4ij.installation.LanguageServerInstallerBase;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+public class BamlLanguageServerInstaller extends LanguageServerInstallerBase {
+
+    private static final Path BAML_CACHE_DIR = Path.of(System.getProperty("user.home"), ".baml");
+    private static final Path BREADCRUMB_PATH = Path.of(System.getProperty("user.home"), ".baml", "baml-cli-0.89.0-from-github");
+    private static final Path TAR_GZ_PATH = Path.of(System.getProperty("user.home"), ".baml", "baml-cli-0.89.0-from-github.tar.gz");
+
+    @Override
+    protected boolean checkServerInstalled(@NotNull ProgressIndicator indicator) {
+        progress("Checking if the language server is installed...", indicator);
+        ProgressManager.checkCanceled();
+        return Files.exists(BREADCRUMB_PATH);
+    }
+
+    @Override
+    protected void install(@NotNull ProgressIndicator indicator) throws Exception {
+        progress("Downloading server components...", 0.25, indicator);
```

Comment:
As you download from github releases I think you could use my current work with installation (not released) https://github.com/redhat-developer/lsp4ij/blob/main/docs/UserDefinedLanguageServerTemplate.md#github-asset-download

I need to write more doc

Reply:
No sorry, I need to fix some instal issues again. I hope I could do the release next week.